packages/client/src/types.js

```diff
@@ -14,8 +14,10 @@ export let getTypePropOrError = _.curry(
 )
 
 export let runTypeFunctionOrDefault = _.curry(
-  (defaultFn, types, prop, node, extend) =>
-    (getTypeProp(types, prop, node) || defaultFn)(node, extend)
+  (defaultFn, types, prop, node, extend) => {
+    let typeFunction = getTypeProp(types, prop, node)
+    return (_.isFunction(typeFunction) ? typeFunction : defaultFn)(node, extend)
```

Comment:
When would typeFunction be defined but not be a function? Is it because hasResult is the prop name and the function name? If so, it might make sense to change the name of one of them 🤔